workers/handshake-manager/src/manager/matching/external_engine.rs

```diff
@@ -228,8 +264,90 @@ impl HandshakeExecutor {
             match_res,
             response_topic,
         );
+        self.enqueue_settlement_task(task, options).await
+    }
+
+    // --- Bounded Match Handler --- //
+
+    /// Handle a bounded match result on the given order
+    ///
+    /// The matching engine will sample appropriate bounds for the `base_amount`
+    /// and forward the task to the task driver to create a bundle.
+    async fn handle_bounded_match(
+        &self,
+        order_id: OrderIdentifier,
+        price: TimestampedPrice,
+        match_res: MatchResult,
+        response_topic: String,
+        options: &ExternalMatchingEngineOptions,
+    ) -> Result<(), HandshakeManagerError> {
+        // Build a bounded match result
+        let price_fp = price.as_fixed_point();
+        let (min_amount, max_amount) =
+            self.derive_match_bounds(&order_id, price_fp, match_res.base_amount).await?;
+        let bounded_res = BoundedMatchResult {
+            quote_mint: match_res.quote_mint,
+            base_mint: match_res.base_mint,
+            price: price_fp,
+            min_base_amount: min_amount,
+            max_base_amount: max_amount,
+            direction: match_res.direction,
+        };
+
+        // Create a task to settle the match
+        let wallet_id = self.get_wallet_id_for_order(&order_id).await?;
+        let task = SettleMalleableExternalMatchTaskDescriptor::new(
+            options.bundle_duration,
+            order_id,
+            wallet_id,
+            bounded_res,
+            response_topic,
+        );
+        self.enqueue_settlement_task(task, options).await
+    }
+
+    /// Derive the bounds for a match
+    ///
+    /// Returns a tuple containing `(min_amount, max_amount)`
+    async fn derive_match_bounds(
+        &self,
+        order_id: &OrderIdentifier,
+        price: FixedPoint,
+        match_amt: Amount,
+    ) -> Result<(Amount, Amount), HandshakeManagerError> {
+        let wallet = self
+            .state
+            .get_wallet_for_order(order_id)
+            .await?
+            .ok_or_else(|| HandshakeManagerError::state(ERR_NO_WALLET))?;
+        let order = wallet
+            .get_order(order_id)
+            .ok_or_else(|| HandshakeManagerError::state(ERR_NO_ORDER))?
+            .clone();
+
+        let capitalizing_balance = wallet.get_balance_for_order(&order).unwrap_or_default();
+        let order_min = order.min_fill_size;
+        let order_max = compute_max_amount(&price, &order.into(), &capitalizing_balance);
 
-        let descriptor = TaskDescriptor::from(task);
+        let min_amount = mul_amount_f64(match_amt, 1.0 - MAX_MALLEABLE_RANGE);
+        let max_amount = mul_amount_f64(match_amt, 1.0 + MAX_MALLEABLE_RANGE);
+
+        // Clamp the amounts to the order's min and max
+        let min_amount = u128::max(min_amount, order_min);
+        let max_amount = u128::min(max_amount, order_max);
```

Comment:
Haven't looked yet but I'm assuming we have safety checks / error messages in the API layer to let a customer know when their bounds are too wide.

Reply:
Nvm, I see now that the API layer doesn't see the change in base amount.